test/TestUtilities/GraphicsUtil.cpp

```diff
@@ -16,4 +17,14 @@ namespace sf
 
         return stream.str().c_str();
     }
+
+    doctest::String toString(const sf::Transform& transform)
+    {
+        std::ostringstream stream;
+        stream << transform.getMatrix()[0] << ", " << transform.getMatrix()[4] << ", " << transform.getMatrix()[12] << ", ";
+        stream << transform.getMatrix()[1] << ", " << transform.getMatrix()[5] << ", " << transform.getMatrix()[13] << ", ";
+        stream << transform.getMatrix()[3] << ", " << transform.getMatrix()[7] << ", " << transform.getMatrix()[15];
+
+        return stream.str().c_str();
```

Comment:
I found it a bit weird that their `String` class didn't accept conversion from `std::string`, but [it's indeed the case](https://github.com/doctest/doctest/blob/a384864346f48c7ff5608051c176e7bd505712ae/doctest/parts/doctest_fwd.h#L487-L543).

Possibly it's to reduce header dependencies.

Reply:
I also found that odd. Once #1944 is merged I'll submit another PR that replaces all these doctest formatters with `operator <<` which Vittorio agrees is a better choice.